packages/web-fragments/src/gateway/pagesMiddleware.ts

```diff
@@ -63,12 +64,39 @@ export function getPagesMiddleware(
           element(element) {
             const scriptType = element.getAttribute('type');
             if(scriptType) {
-                element.setAttribute('data-script-type', scriptType);
+              element.setAttribute('data-script-type', scriptType);
             }
             element.setAttribute('type', 'inert');
           }
         });
-        const fragmentRes = scriptRewriter.transform(await fetch(upstreamUrl, { ...request }));
+
+        const fragmentReq: RequestInfo = { ...request, url: upstreamUrl };
+        let fragmentRes: Response;
+        let fragmentFailedResOrError: Response | unknown | null = null;
+        try {
+          fragmentRes = scriptRewriter.transform(await fetch(fragmentReq));
```

Comment:
should we skip transforming if we get an error response?

Reply:
oh yeah good call 🙂